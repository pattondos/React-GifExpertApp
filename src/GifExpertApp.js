import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Cousins']);

    return (
        <>
            <h1>GifExpertApp</h1>
            <hr />
            <AddCategory setCategories={setCategories} />
            <hr />
            <ol>
                {categories.map(
                    category =>
                        <GifGrid
                            key={category}
                            category={category}
                        />
                )
                }
            </ol>
        </>
    );
}

export default GifExpertApp;