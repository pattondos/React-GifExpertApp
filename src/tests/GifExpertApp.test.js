import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en <GifExpertApp />', () => {

    test('Debe Contener los elementos', () => {

        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar una lista de categorias', () => {

        const categories = ['love', 'friend'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    })

});
