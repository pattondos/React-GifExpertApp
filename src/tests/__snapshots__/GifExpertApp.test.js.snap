// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe Contener los elementos 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="love"
      key="love"
    />
    <GifGrid
      category="friend"
      key="friend"
    />
  </ol>
</Fragment>
`;
